.github/workflows/test-print-file-to-step-summary.yml

```diff
@@ -69,18 +81,16 @@ jobs:
           cat << 'EOF' > expected_file
```

Comment:
can we create the expected file separately and use this step just to check the diff.
like `name: Write stub Accumulated report` step and can use `run: diff "$OUTPUT_FILE" "$EXPECTED_ACCUMULATED_FILE`"
This will be consistent with the other code in this test workflow , this make it simple and easy to understand

Reply:
This is not necessary, as for all these test cases creating the expected file is most of the testing step. If we move the creation of the expected file to a sepate step for each test case, then we create a lot of additional steps and the steps checking the diff would all have only the one line `diff "$OUTPUT_FILE" "$EXPECTED_FILE"`.

The way it's organised now makes more sense.